### DISCUSSION TITLE
How to optimize calculation  (Skopt vs grid)

### DISCUSSION THREAD

**Discussion by Benouare** [kernc/backtesting.py#446] [Q&A] [answered]
Hi, 
I would like to use Bt with the optimize method. 
When I use it with something like 1000 possibilities, it's take some time, but that's not too/so long. 
But some times, I would like to use it with 300k possibilities/calculations. 

So a part of the solution could(must) be to reduce the number of calculations, but an other part could be to have a big cpu or something like that to be able to make the calculation. 

Currently, i am running backtesting on a Imac 2K10/11. I dont remmeber the exact configuration, but I think it's a I5 with 16 Go of ram. (edit : 2.7 i5 16go Ram)

So my question is : what kind of hardware should i have to optimise my calculation? 

Cheers

**Comment by kernc** [comment, marked as answer]
> with 300k possibilities/calculations

Foremost, with as wide hyperparameter space, I'd set [`Backtest.optimize(..., method='skopt')`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.optimize) as showcased in [parameter heatmap / optimization tutorial](https://kernc.github.io/backtesting.py/doc/examples/Parameter%20Heatmap%20&%20Optimization.html#Model-based-optimization).

**Reply by Benouare** [reply]
That's a good tip! Thx
